[Feature] Runner 구현

Ellen99999999 opened the pull request:
변경점
1. script 쪽에서 executeAll 을 만들었음.shell shell_script.txt 를 입력하면 해당 파일에 적힌 tc 를 모조리 실행할 수 있도록 함
2. release mode 에서 ProcessParseInvalid 이후에 검증된 command 는 NO_ERROR 처리해서 release mode exe 에서도 실행될 수 있도록 했음.
3. shell_script.txt 에 현재 까지 작성된 script tc 목록을 추가하였음

검증

[ v] Build

[ v] Regression Test Pass

[ v] 해당 기능 TC

Ellen99999999 committed:
[combine]
bug fix. 복붙으로 만들다보니 불필요한 객체생성된 것 삭제

exe file 추가(릴리즈 모드)

[Feature] shell runner 실행 방식 수정

변경점 : main.cpp() 안에서 인풋 아규먼트가 txt 로 추가가 더 들어오는 경우는 RUN() 을 돌지 않고 그냥 전체 script 실행을 하는 ExecuteAll() 호출 방식으로 변경함
추가로 big loop 안에서 exit 가 먹히지 않는 오류 같이 수정함
release mode big loop 안에서 5회 테스트용 수행 코드는 _DEBUG 로 처리함

검증

[ v] Build

[ v] Regression Test Pass

[ v] 해당 기능 TC + 릴리즈 모드에서 커맨드 직접 수행해봄

[Feature] Runner 구현

변경점
1. script 쪽에서 executeAll 을 만들었음.shell shell_script.txt 를 입력하면 해당 파일에 적힌 tc 를 모조리 실행할 수 있도록 함
2. release mode 에서 ProcessParseInvalid 이후에 검증된 command 는 NO_ERROR 처리해서 release mode exe 에서도 실행될 수 있도록 했음.
3. shell_script.txt 에 현재 까지 작성된 script tc 목록을 추가하였음

검증

[ v] Build

[ v] Regression Test Pass

[ v] 해당 기능 TC

Ellen99999999 committed:
[combine]
bug fix. 복붙으로 만들다보니 불필요한 객체생성된 것 삭제

exe file 추가(릴리즈 모드)

[Feature] shell runner 실행 방식 수정

변경점 : main.cpp() 안에서 인풋 아규먼트가 txt 로 추가가 더 들어오는 경우는 RUN() 을 돌지 않고 그냥 전체 script 실행을 하는 ExecuteAll() 호출 방식으로 변경함
추가로 big loop 안에서 exit 가 먹히지 않는 오류 같이 수정함
release mode big loop 안에서 5회 테스트용 수행 코드는 _DEBUG 로 처리함

검증

[ v] Build

[ v] Regression Test Pass

[ v] 해당 기능 TC + 릴리즈 모드에서 커맨드 직접 수행해봄

[Feature] Runner 구현

변경점
1. script 쪽에서 executeAll 을 만들었음.shell shell_script.txt 를 입력하면 해당 파일에 적힌 tc 를 모조리 실행할 수 있도록 함
2. release mode 에서 ProcessParseInvalid 이후에 검증된 command 는 NO_ERROR 처리해서 release mode exe 에서도 실행될 수 있도록 했음.
3. shell_script.txt 에 현재 까지 작성된 script tc 목록을 추가하였음

검증

[ v] Build

[ v] Regression Test Pass

[ v] 해당 기능 TC

Ellen99999999 commented on TeamBest_SSD_Shell/script_executor.h:
```diff
@@ -8,31 +8,65 @@
+		return false;
+	}
+
+	bool ExecuteAll(const std::string& filename) { 
```
추가한 함수는 ExecuteAll() 만 있는데 나머지가 들여쓰기가 바뀌면서 다 바뀐것 처럼 보이는군요 ㅋㅋ

flowinto10 commented on TeamBest_SSD_Shell/script_executor.h:
```diff
@@ -8,31 +8,65 @@
-        return false;
-    }
+	ScriptExcutor() {
+		registerCommand("1_", std::unique_ptr<ScriptCommand>(new FullWRiteAndReadCompare()));
```
각 TC Class는 모두 생성해서 등록해두고, txt에 등록된 TC만 수행되도록 하는건가요?

flowinto10 commented on TeamBest_SSD_Shell/SSD_Shell.h:
```diff
@@ -20,7 +20,11 @@ enum Command {
-	SCRIPT_EXECUTE = 7
+	SCRIPT_EXECUTE = 7,
+	ERASE = 8,
+	ERASE_RANGE = 9,
```
ERASE는 원석님 코드리뷰 진행중이신데 미리 반영되어있네요. 혹시 코드가 꼬인건가요?

Ellen99999999 replied:
네 꼬일까봐 미리 반영한거에요 ㅎㅎ

flowinto10 commented on TeamBest_SSD_Shell/script_executor.h:
```diff
@@ -8,31 +8,65 @@
+				// 공백 제거
+				inputCommand.erase(std::remove_if(inputCommand.begin(), inputCommand.end(), ::isspace), inputCommand.end());
+
+				ScriptExcutor executor;
```
해당 method가 SriptExcutor 클래스 내에 엤는데, 별도로 생성해서 수행해야 하는건가요?

Ellen99999999 replied:
강사님꼐 다시 문의 했더니 별도 생성이 맞군요 새로 만들겠습니다. ㅠㅠㅠㅠㅠㅠㅠ

wonseokkim777 reviewed (approved):
cout 은 차후 logger 함수로 치환하면 될것 같습니다.

flowinto10 commented on TeamBest_SSD_Shell/SSD_Shell.cpp:
요구사항에 보면, Script.txt 는 shell의 실행과 함께 받도록 되어 있어서,
main() 함수에서 argument 를 받도록 수정이 필요할 것 같은데. 별도로 구현 예정이신가요?

Ellen99999999 replied:
네네 다시 구현을 하겠습니다. 큰일날 뻔 했네용 ㅠㅠㅠㅠㅠㅠ

R1NRINRIN commented on TeamBest_SSD_Shell/SSD_Shell.cpp:
```diff
@@ -131,15 +132,22 @@ bool SSDShell::ExcuteCommand(ParsingResult command) {
+	case SCRIPT_EXECUTE:
+	{
+		cout << "Script Execute" << endl;
+		ScriptExcutor* scriptExcutor = new ScriptExcutor();
```
필수적인 것은 아니지만 여러 case에서 사용된다면 switch문 밖에서 선언하는 것도 좋을 것 같습니다.
강의 시간에 배운 std::shared_ptr 도 활용해볼 수 있을 것 같습니다.

R1NRINRIN reviewed (approved):
가독성이 좋은 코드입니다. 고생 많으셨습니다

Ellen99999999 commented on TeamBest_SSD_Shell/main.cpp:
```diff
@@ -5,16 +5,29 @@
+	}
+	else if (argc == 2) {
+		ScriptExcutor* scriptExcutor = new ScriptExcutor();;
+		scriptExcutor->ExecuteAll(argv[1]);
```
여기서 shell.exe script.txt 처럼 실행하면 동작하게 바꿨어요

Ellen99999999 commented on TeamBest_SSD_Shell/SSD_Shell.cpp:
```diff
@@ -1,11 +1,16 @@
 	while (loopCount < 5) {
+#else
+	while (true) {
+#endif
```
big loop 를 릴리즈 모드일 때는 loop count 제약 없도록 수정합니다.

flowinto10 replied:
감사합니다. 이건 제가 정리했어야 하는데.ㅎㅎ

Ellen99999999 commented on TeamBest_SSD_Shell/SSD_Shell.cpp:
```diff
@@ -112,34 +117,38 @@ bool SSDShell::ExcuteCommand(ParsingResult command) {
+			cout << "Exit" << endl;
+			ret = true;
+			break;
+		case SCRIPT_EXECUTE:
```
script.txt 는 아니고 개별 script (예: 1_어쩌고...) 동작하는거 누락된거 추가요

Ellen99999999 commented on TeamBest_SSD_Shell/SSD_Shell.cpp:
```diff
@@ -231,21 +240,35 @@ bool SSDShell::ProcessParseInvalid(std::string command) {
 			return false;
 		}
 	}
+	else if (parsingresult.command == SCRIPT_EXECUTE) {
```
script.txt 는 아니고 개별 script (예: 1_어쩌고...) 동작하는거 누락된거 추가요

Ellen99999999 commented on TeamBest_SSD_Shell/SSD_Shell.cpp:
```diff
@@ -297,7 +320,8 @@ bool SSDShell::UpdateCommand(std::string cmd) {
 	else { return false; }
-	
+
+	parsingresult.invalidtype = NO_ERROR;
```
이게 없으면 모든 파싱된 커맨드들이 invalidtype 이 이상한게 박혀서 실행이 안되더라구요.
수정합니다.

Ellen99999999 commented on x64/Release/ssd.exe:
얘는 릴리즈모드의 shell.exe 가 실행되기 위해 필요해서 넣어놨습니다.

flowinto10 reviewed (approved):
고생하셨습니다. 
테스트 스크립트를 쉽게 돌릴수 있겠네요.

Ellen99999999 committed:
Merge branch 'feature/shell_runner_2' of https://github.com/flowinto10/CRA_TeamBest_SSDShell into feature/shell_runner_2

Ellen99999999 committed:
[feature] script runner 릴리즈/디버그 모드에서 동작하도록 변경
변경점 :
리베이스 후 안되는거 수정 - 릴리즈 모드 c++17 되도록 변경 (빌드에러 수정)

검증

[ v] Build

[ v] Regression Test Pass

[ v] 해당 기능 TC
